### PR TITLE
fix(sct_runner): upgrade runner image to Ubuntu 22.04

### PR DESCRIPTION
**Pull request by enaydanov**
Modern Scylla images built on top of Ubuntu 22.04 and in result we have compatibility problems when try to run Scylla' Docker image on SCT Runner.

For example:

```
$ docker run --entrypoint java scylladb/scylla-nightly -version [0.003s][warning][os,thread] Failed to start thread "GC Thread#0" - pt... 
#
# There is insufficient memory for the Java Runtime Environment to continue. 
# Cannot create worker GC thread. Out of system resources. 
# An error report file with more information is saved as: 
# //hs_err_pid1.log
```
This particular problem w/ Java makes it impossible to run c-s on SCT Runner using Scylla' Docker image.

See https://stackoverflow.com/a/72841934

Also, we need to upgrade base image for SCT because of too old sudo package.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [x] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by enaydanov** [comment]
I built and copied new SCT Runner image across all supported clouds and regions.

Staging longevity run: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/evgeniy/job/longevity-100gb-4h-test/3/ -- not finished yet, but looks good.

@scylladb/qa-maintainers please, don't merge before I change version of hydra to v1.29 and push new image to the Docker Hub.

**Comment by fruch** [comment]
@enaydanov we just need to upgrade the docker version on the runners. the upgrade to newer Ubuntu might  introduce other issues (like ssh connectivity)

Please make sure you created the images across all GCP projects (we now have 3 of them)

**Comment by enaydanov** [comment]
Israel and I decided to postpone the merge of this PR till the end of HNY holidays because of high risks.

**Comment by fruch** [comment]
@enaydanov also I'm not sure we need to update the hydra image in this PR, it has nothing todo with the issue at hand.

**Comment by enaydanov** [comment]
> @enaydanov also I'm not sure we need to update the hydra image in this PR, it has nothing todo with the issue at hand.

@fruch there is the sudo compatibility issue between buster and 22.04: sudo 1.9.1 slightly changed syntax of config file.
And since we mount /etc/sudoers and /etc/sudoers.d to a SCT Docker container we have following problems with running buster based SCT image on Ubuntu 22.04 host:

```
 Command: 'sudo stat -c "%U:%G" /tmp/tmp.DkNgpsHKBi'
 
 Exit code: 1
 
 Stdout:
 
 
 
 Stderr:
 
 >>> /etc/sudoers: syntax error near line 54 <<<
 sudo: parse error in /etc/sudoers near line 54
 sudo: no valid sudoers sources found, quitting
 sudo: unable to initialize policy plugin
```

So, it's a part of solution.

**Comment by enaydanov** [comment]
This PR passed all provision tests: AWS, GCP, Azure, Docker.

The new runner image (sct-runner-1.6) was created in following clouds and regions/projects:

* AWS
  * us-east-1 (ami-0b3810358d098341b)
  * us-east-2 (ami-0bf04c68613816bc7)
  * us-west-1 (ami-05027e39080a0d4ad)
  * us-west-2 (ami-0d65455788c01993a)
  * ap-south-1 (ami-0266ba4b5ea47fe93)
  * ap-northeast-2 (ami-0e7c22e40a3f1447c)
  * ap-southeast-1 (ami-029808bd8cf24539a)
  * ap-southeast-2 (ami-0c6f1be67ca847034)
  * ap-northeast-1 (ami-04bd65cc0bd14f3d2)
  * ca-central-1 (ami-064222902bc744f70)
  * eu-central-1 (ami-0fe9c488c4c388c07)
  * eu-west-1 (ami-0102cef538ddd92e8)
  * eu-west-2 (ami-0e019de6d67c7c5ce) <-- master image for all other regions
  * eu-west-3 (ami-02dcf9fd0d1c6d916)
  * eu-north-1 (ami-09b97156776627902)
  * sa-east-1 (ami-036ca48d5a9f5c046)
* GCP
  * skilled-adapter-452
  * sct-project-1
  * local-ssd-latency
* Azure
  * eastus

Don't see any SSH connectivity problems with SCT runners started from these images.

**Review by soyacz — APPROVED** [review]

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**
